File: cinemaapp/src/app/interfaces/film.ts
export class Film {
    public idFilm: number;
   public nume: string;
   public gen: string;
   public regizor: string;
   public dataLansare: string;
   public notaImdb: number ;
   public durata: number ;
   public rating: number ;
  
  
   constructor() {
      this.idFilm = null as any;
     this.nume = '';
     this.gen = '';
     this.regizor = '';
     this.dataLansare = '';
     this.notaImdb = null as any;
     this.durata = null as any;
     this.rating = null as any;

   }

}

File: cinema/src/main/java/com/example/cinema/model/Film.java
package com.example.cinema.model;


import jakarta.persistence.*;
import lombok.*;

import java.util.Date;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Table(name = "film")
public class Film {


@Id
//@GeneratedValue(strategy = GenerationType.IDENTITY)
//@Column(nullable = false, updatable = false)
@GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_film")
    private Long idFilm;

    private String nume;
    private String gen;
    private String regizor;
    private String dataLansare;
    private Integer notaImdb;
    private Integer rating;
    private Integer durata;
}
